http/server.ts

```diff
@@ -622,8 +543,12 @@ export async function serve(
   handler: Handler,
   options: ServeInit = {},
 ): Promise<void> {
-  const addr = options.addr ?? ":8000";
-  const server = new Server({ addr, handler, onError: options.onError });
+  const server = new Server({
+    port: options.port ?? 8000,
+    hostname: options.hostname ?? "0.0.0.0",
+    handler,
+    onError: options.onError,
```

Comment:
An aside to this PR (highlighted due to formatting changes) - curious that have the `onError` option for `serve()` but not `serveTls()`

Reply:
Right. That was a known issue and addressed at #1679